dq.go

```diff
@@ -0,0 +1,215 @@
+package main
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/TykTechnologies/tyk-cluster-framework/client"
+	"github.com/TykTechnologies/tyk-cluster-framework/encoding"
+
+	"github.com/jeffail/tunny"
+
+	"github.com/Sirupsen/logrus"
+
+	"github.com/TykTechnologies/dq"
+	"github.com/TykTechnologies/tyk/apidef"
+)
+
+var DQFlusherPool *tunny.WorkPool = tunny.CreatePoolGeneric(10)
+var QuotaHandler *dq.DistributedQuota
+
+type GetLeaderStatusFunc func() bool
+
+func getDQTopic() string {
+	topic := "tyk.dq"
+	if config.DBAppConfOptions.NodeIsSegmented {
+		if len(config.DBAppConfOptions.Tags) > 0 {
+			tags := strings.Join(config.DBAppConfOptions.Tags, ".")
+			topic = topic + "." + tags
+		}
+	}
+
+	return topic
+}
+
+func dqErrorHandler(e error) {
+	log.WithFields(logrus.Fields{
+		"prefix": "main.DQ",
+	}).Error(e)
+}
+
+var dummyAPISpec = APISpec{APIDefinition: &apidef.APIDefinition{SessionLifetime: 0}}
+
+func dqFlusher(d map[string]*dq.Quota) error {
+	for k, v := range d {
+		DQFlusherPool.SendWork(func() {
+			// Ignore deleted flags
+			if v.Delete {
+				log.Warning("Key has been tagged for deletion, not writing counter")
+				return
+			}
+
+			// We will track all the session handlers for this key
+			processedSpecs := map[SessionHandler]struct{}{}
+
+			// Let's go through all the API IDs in the metadata so we capture all the handlers
+			apis := v.Meta.(map[string]interface{})["Apis"]
+
+			// I hate this
```

Comment:
(to clarify - this is because in the JSON spec all numbers are float64, and if Go is just given a map instead of a struct, it defaults to float64 for compatibility)